src/overlay/PeerManager.h

```diff
@@ -87,9 +86,13 @@ class PeerManager
     void ensureExists(PeerBareAddress const& address);
 
     /**
-     * Update type of peer associated with given address.
+     * Update type of peer associated with given address. This function takes
+     * desired peer type, and whether the preferred type is definitely known (in
+     * some cases it is unknown whether a peer is preferred or not).
+     * Depending on the peer type stored in the DB, a new type is determined.
      */
-    void update(PeerBareAddress const& address, TypeUpdate type);
+    void update(PeerBareAddress const& address, PeerType desiredType,
```

Comment:
`desiredType` doesn't seem accurate, I think it's more an `observedType`

Reply:
done